Software/Rig/Rig.ino

```diff
@@ -0,0 +1,339 @@
+    if (!isPuffing && ((currentTime - puffStopTime) > INTER_PUFF_PAUSE_TIME)) {
+      digitalWrite(PIN_AIR_PUFF, HIGH);
+      print("Puff start");
+      puffStartTime = currentTime;
+    }
+    if (isPuffing && ((currentTime - puffStartTime) > AIR_PUFF_DURATION)) {
+      digitalWrite(PIN_AIR_PUFF, LOW);
+      print("Puff stop");
+      puffStopTime = currentTime;
+    }
+  } else {
+    // Makes sure we are not puffing
+    if (isPuffing) {
+      digitalWrite(PIN_AIR_PUFF, LOW);
+      print("Puff stop, catch block");
+      puffStopTime = currentTime;
+    }
+  }
+}
+void flushAirPuffMetaData() {
+  puffStartTime = __LONG_MAX__;
+  puffStopTime = 0;
+
+  digitalWrite(PIN_AIR_PUFF, LOW);
+  print("Puff stop via trial flush");
+}
+
+
+/* POSITIVE SIGNAL
+ *
+ */
+void checkPositiveSignal() {
+  trialTime = millis() - trialStartTime;
+  if (!positiveSignalPlaying && (trialTime > AUDITORY_START) && (trialTime < (AUDITORY_STOP - AUDITORY_BUFFER))) {
+    // Should only hit this code once per trial.
+    // The buffer tries to prevent entering a second time, since:
+    //
+    //    POSITIVE_DURATION = AUDITORY_STOP - AUDITORY_START
+    //
+    tone(PIN_TONE_POSITIVE, POSITIVE_FREQUENCY, POSITIVE_DURATION);
+    positiveSignalPlaying = true;
+    positiveSignalStart = trialTime;
+    print("Positive signal start");
+  } else if (positiveSignalPlaying && ((trialTime - positiveSignalStart) > POSITIVE_DURATION)) {
+    noTone(PIN_TONE_POSITIVE);
+    positiveSignalPlaying = false;
+    print("Positive signal stop");
+  }
+}
+void flushPositiveSignalMetaData() {
+  positiveSignalPlaying = false;
+  positiveSignalStart = __LONG_MAX__;
+  noTone(PIN_TONE_POSITIVE);
+  print("Positive signal stop via trial flush");
+}
+
+
+/* NEGATIVE SIGNAL
+ * This needs to pulse the tone, unlike positive.
+ *
+ */
+void checkNegativeSignal() {
+  trialTime = millis() - trialStartTime;
+  if (trialTime > AUDITORY_START) && (trialTime < (AUDITORY_STOP - AUDITORY_BUFFER)) {
+    if (!negativeSignalPlaying && ((trialTime - negativeSignalStop) > NEGATIVE_CYCLE_DURATION)) {
+      tone(PIN_TONE_NEGATIVE, NEGATIVE_FREQUENCY, NEGATIVE_PULSE_DURATION);
+      negativeSignalPlaying = true;
+      negativeSignalStart = trialTime;
+      print("Negative signal start");
+    } else if (negativeSignalPlaying && ((trialTime - negativeSignalStart) > NEGATIVE_PULSE_DURATION)) {
+      // Calling `noTone` is redundant with letting `tone` stop a signal
+      // of length `NEGATIVE_PULSE_DURATION` but we want to be consistent
+      // in how we treat and print tones
+      noTone(PIN_TONE_NEGATIVE);
+      negativeSignalPlaying = false;
+      negativeSignalStop = trialTime;
+      print("Negative signal stop");
+    }
+  }
+}
+void flushNegativeSignalMetaData() {
+  negativeSignalPlaying = false;
+  negativeSignalStart = __LONG_MAX__;
+  negativeSignalStop = 0;
+  noTone(PIN_TONE_NEGATIVE);
+  print("Negative signal stop via trial flush")
+}
+
+
+/* PRINTING HELPER FUNCTIONS
```

Comment:
Can ignore this other than noting that it outputs like:

```text
trialNumber: millis: print statement
```

And with the option of providing a string (like a variable name) and value with `vprint`:

```text
trialNumber: millis: statement: value
```

The idea is, this format can be read in another setting (like Colab) to visualize output, both to confirm experiment setup by reading start/stop of different components, and to see when signals happen